bpf/tc/attach.go

```diff
@@ -472,34 +496,35 @@ func RemoveQdisc(ifaceName string) error {
 	if !hasQdisc {
 		return nil
 	}
-	_, err = ExecTC("qdisc", "del", "dev", ifaceName, "clsact")
-	if err != nil {
-		return fmt.Errorf("failed to remove qdisc from interface '%s': %w", ifaceName, err)
+	return libbpf.RemoveQDisc(ifaceName)
+}
+
+func RemoveTCOpts(ifaceName string) {
+	optsLock.Lock()
+	defer optsLock.Unlock()
+	if optsMap != nil {
+		key := ifaceName + "_" + "ingress"
+		delete(optsMap, key)
+		key = ifaceName + "_" + "egress"
+		delete(optsMap, key)
 	}
-	return nil
 }
 
 func (ap *AttachPoint) ProgramID() (string, error) {
 	logCtx := log.WithField("iface", ap.Iface)
 	logCtx.Info("Finding TC program ID")
-	out, err := ExecTC("filter", "show", "dev", ap.Iface, string(ap.Hook))
-	if err != nil {
-		return "", fmt.Errorf("failed to find TC filter for interface %v: %w", ap.Iface, err)
-	}
-	logCtx.Infof("out:\n%v", out)
-
-	progName := ap.ProgramName()
-	for _, line := range strings.Split(out, "\n") {
-		if strings.Contains(line, progName) {
-			re := regexp.MustCompile(`id (\d+)`)
-			m := re.FindStringSubmatch(line)
-			if len(m) > 0 {
-				return m[1], nil
+	key := ap.Iface + "_" + string(ap.Hook)
+	if optsMap != nil {
+		optsLock.RLock()
+		defer optsLock.RUnlock()
+		if val, ok := optsMap[key]; ok {
```

Comment:
If the access to the optsMap was wrapped in functions like the `GetTCOpts` this could read just `if ExistsTCOpts() { ... }` or `if TCOptsExists(...)` ... (also you would not hold the lock for the rest of the function - i guess it does not matter here though)